velox/connectors/hive/storage_adapters/s3fs/S3Util.h

```diff
@@ -80,22 +81,62 @@ inline bool isS3File(const std::string_view filename) {
       isOssFile(filename) || isCosFile(filename) || isCosNFile(filename);
 }
 
+// Currently,Amazon S3,Alibaba OSS and Tencent COS all support access through
+// s3fs in velox, but each object storage service has different restrictions on
+// bucket names. Here, we have taken the intersection of the bucket name
```

Comment:
@kevincmchen Why do we have to take the intersection? Can't you define the validator for each storage service since the scheme is different?

Reply:
Taking the intersection or union of the rules is not going to help catch real issues.

Reply:
got it, thanks!  I rewrite this pr based on your suggestion.  please help review it! @majetideepak